src/main/java/com/atlassian/db/replica/spi/DatabaseCluster.java

```diff
@@ -0,0 +1,10 @@
+package com.atlassian.db.replica.spi;
+
+import com.atlassian.db.replica.api.Database;
+
+import java.sql.SQLException;
+import java.util.stream.Stream;
+
+public interface DatabaseCluster {
```

Comment:
DatabaseCluster should contain all databases, MAIN + REPLICAS or Writer with Readers

Reply:
Other than that, databaseEngine - as it might be Aurora, might be not.

Reply:
> DatabaseCluster should contain all databases, MAIN + REPLICAS or Writer with Readers

No use for the main database yet.

> Other than that, getDatabaseEngine - as it might be Aurora, might be not.

What use case do you mean? The implementation defines the database/database engine.

Reply:
Ok, to be fair, the first thing I did I looked into TCS into customer code and looked what it know about DatabaseCluster.

you're right with focusing on API usecases first.